ext/xml/xml.c

```diff
@@ -1442,15 +1442,15 @@ PHP_FUNCTION(xml_parser_set_option)
 
 			enc = xml_get_encoding((XML_Char*)Z_STRVAL_P(val));
 			if (enc == NULL) {
-				php_error_docref(NULL, E_WARNING, "Unsupported target encoding \"%s\"", Z_STRVAL_P(val));
-				RETURN_FALSE;
+				zend_argument_value_error(3, "is an unsupported target encoding");
+				RETURN_THROWS();
 			}
 			parser->target_encoding = enc->name;
 			break;
 		}
 		default:
-			php_error_docref(NULL, E_WARNING, "Unknown option");
-			RETURN_FALSE;
+			zend_argument_value_error(2, "must be one of PHP_XML_OPTION_CASE_FOLDING, PHP_XML_OPTION_SKIP_TAGSTART, PHP_XML_OPTION_SKIP_WHITE, PHP_XML_OPTION_TARGET_ENCODING");
```

Comment:
```suggestion
			zend_argument_value_error(2, "must be a PHP_XML_OPTION_* constant");
```
or similar. I think we should not list out all possible values in cases such as these.

Reply:
I like these suggestion so went with them!